src/wallet/wallet.cpp

```diff
@@ -3027,9 +3027,27 @@ int64_t CWallet::GetOldestKeyPoolTime()
     if (setKeyPool.empty())
         return GetTime();
 
-    // load oldest key from keypool, get time and return
     CKeyPool keypool;
     CWalletDB walletdb(strWalletFile);
+
+    if (IsHDEnabled() && CanSupportFeature(FEATURE_HD_SPLIT))
+    {
+        // if HD & HD Chain Split is enabled, response max(oldest-internal-key, oldest-external-key)
+        int64_t now = GetTime();
+        int64_t oldest_external = now, oldest_internal = now;
+
+        for(const int64_t& id : setKeyPool)
+        {
+            if (!walletdb.ReadPool(id, keypool))
+                throw runtime_error(std::string(__func__) + ": read failed");
+            if (keypool.fInternal && keypool.nTime < oldest_internal)
+                oldest_internal = keypool.nTime;
+            else if (!keypool.fInternal && keypool.nTime < oldest_external)
+                oldest_external = keypool.nTime;
+        }
+        return std::max(oldest_internal, oldest_external);
+    }
+    // load oldest key from keypool, get time and return
```

Comment:
Could you add a comment mentioning why in this case it is safe to assume key with the lowest index is the oldest key, while in the above case, you need to loop through the keys to find the oldest one. It's not clear (at least to me) why FEATURE_HD_SPLIT would affect key order.